drift.py

```diff
@@ -35,31 +36,29 @@ def analyze(self, commit):
                 # If concretization successful check the resulting concrete specs.
                 if out.returncode == SUCCESS:
                     concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
-                    if abstract_spec in self.last and self.last[abstract_spec] != concrete_spec:
-                        # Construct Result
-                        diff = spack.cmd.diff.compare_specs(self.last[abstract_spec], concrete_spec, colorful=False)
-                        result = Result(
-                            abstract_spec[:verToken],
-                            abstract_spec[verToken+1:],
-                            commit.hash,
-                            # Waiting for spack diff to complete tags.
-                            diff['b_not_a'],
-                            str(commit.author_date))
-                        # Send result to drift-server
-                        send(result)
-                    # Save concrete spec as last spec
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last or self.last[abstract_spec] == concrete_spec:
+                        continue
+                    # If the spec is different check what is different and save as tags.
+                    diff = spack.cmd.diff.compare_specs(
+                        self.last[abstract_spec],
+                        concrete_spec,
+                        colorful=False)
+                    tags = diff['b_not_a']
+                    # Save concrete spec as last spec.
                     self.last[abstract_spec] = concrete_spec
                 # Mark failing concretization points.
                 else:
-                    # Construct Result
-                        result = Result(
-                            abstract_spec[:verToken],
-                            abstract_spec[verToken+1:],
-                            commit.hash,
-                            [("concretization-failed","")],
-                            str(commit.author_date))
-                        # Send result to drift-server
-                        send(result)
+                    tags = [("concretization-failed","")]
+                # Construct Result
+                result = Result(
+                    abstract_spec[:verToken],
+                    abstract_spec[verToken+1:],
+                    commit.hash,
+                    tags,
+                    str(commit.author_date))
+                # Send result to drift-server
+                send(result)
```

Comment:
unless your send function handles this parsing!

Reply:
yep I spoke too soon, it does, lol. So the tags should just be a flat list, and you can use the one liner I showed above if you want to reduce the complexity of send(), which currently has three lines for just the tags parsing.

Reply:
Wow I'd totally forgotten how nice list comprehensions are in Python!